acousticbrainz/views/data.py

```diff
@@ -44,21 +45,44 @@ def view_high_level(mbid):
 
 @data_bp.route("/<uuid:mbid>", methods=["GET"])
 def summary(mbid):
-    lowlevel, highlevel, genres, moods, other = get_summary_data(mbid)
-
-    info = _get_track_info(mbid, lowlevel['metadata'])
+    summary = get_summary_data(mbid)
+    
+    lowlevel = None
+    highlevel = None
+    genres = None
+    moods = None
+    other = None
+    if 'lowlevel' in summary:
+        lowlevel = summary['lowlevel']
+    if 'highlevel' in summary:
+        highlevel = summary['highlevel']
+    if 'genres' in summary:
+        genres = summary['genres']
+    if 'moods' in summary:
+        moods = summary['moods']
+    if 'other' in summary:
+        other = summary['other']
+    
+    info = None
+    info = _get_track_info(mbid, lowlevel['metadata'] if lowlevel else None)
 
     # Tomahawk player stuff
-    if not ('artist' in lowlevel['metadata']['tags'] and 'title' in lowlevel['metadata']['tags']):
+    if not ('artist' in info and 'title' in info):
         tomahawk_url = None
     else:
         tomahawk_url = "http://toma.hk/embed.php?artist={artist}&title={title}".format(
             artist=quote_plus(info['artist'].encode("UTF-8")),
             title=quote_plus(info['title'].encode("UTF-8")))
 
-    return render_template("data/summary.html", lowlevel=lowlevel, highlevel=highlevel,
-                           genres=genres, moods=moods, other=other, mbid=mbid,
-                           info=info, tomahawk_url=tomahawk_url)
+    if lowlevel and info: # When all the data is available
+        return render_template("data/summary.html", lowlevel=lowlevel, highlevel=highlevel,
+                                genres=genres, moods=moods, other=other, mbid=mbid,
```

Comment:
You can just pass ``summary`` there without extracting different parts of it into separate variables. *Template will need to be modified slightly.*